WebView/app/src/main/java/com/android/samples/webviewdemo/MainActivity.kt

```diff
@@ -43,26 +50,71 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
-    /** Instantiate the interface and set the context  */
-    class WebAppInterface(private val mContext: Context) {
-        /** Send a message from the web page  */
-        @JavascriptInterface
-        fun sendMessage(message: String) {
-            val sendIntent: Intent = Intent().apply {
-                action = Intent.ACTION_SEND
-                putExtra(Intent.EXTRA_TEXT, message)
-                type = "text/plain"
-            }
-            val shareIntent = Intent.createChooser(sendIntent, null)
-            startActivity(mContext, shareIntent, null)
+    /**
+     * This method checks to see if WebMessageListener is supported. This method is prefered
```

Comment:
The first sentence should describe the purpose of the method. A better first sentence might be "Injects a JavaScript object which supports a {@code postMessage()} method."

Your other sentences would be a good spot to explain how the origin rules are used, how this chooses between WebMessageListener vs. JavascriptInterface, how the callback works, etc.